app/javascript/controllers/dropdate_picker_controller.js

```diff
@@ -0,0 +1,81 @@
+import { Controller } from "@hotwired/stimulus"
+import flatpickr from "flatpickr"
+
+export default class extends Controller {
+  static targets = ["input", "date", "time", "zone", "showing", "editing"]
+
+  connect() {
+    // localize inputs
+    if (this.inputTarget.value) {
+      const date = new Date(this.inputTarget.value)
+      this.setValue(this.dateTarget, date.toLocaleDateString(), true)
+      this.setValue(this.timeTarget, date.toLocaleTimeString(), true)
+    }
+
+    // listen for changes
+    this.dateTarget.dataset.action = `${this.dateTarget.dataset.action} dropdate-picker#change dropdate-picker#defaultTime`
+    this.timeTarget.dataset.action = `${this.timeTarget.dataset.action} dropdate-picker#change`
+    this.zoneTarget.dataset.action = `${this.zoneTarget.dataset.action} dropdate-picker#change`
+
+    // init flatpickr
+    flatpickr(this.dateTarget, {
+      allowInput: true,
+      dateFormat: "n/j/Y",
+    })
+    flatpickr(this.timeTarget, {
+      allowInput: true,
+      dateFormat: "h:i:S K",
+      enableTime: true,
+      enableSeconds: true,
+      noCalendar: true,
+    })
+  }
+
+  editing(event) {
+    event.preventDefault()
+    this.showingTargets.forEach((el) => el.classList.add("d-none"))
+    this.editingTargets.forEach((el) => el.classList.remove("d-none"))
+  }
+
+  change() {
+    if (this.dateTarget.value && isNaN(new Date(this.dateTarget.value))) {
+      this.dateTarget.classList.add("is-invalid")
+    } else {
+      this.dateTarget.classList.remove("is-invalid")
+    }
+
+    if (this.timeTarget.value && isNaN(new Date(`2020/01/01 ${this.timeTarget.value}`))) {
+      this.timeTarget.classList.add("is-invalid")
+    } else {
+      this.timeTarget.classList.remove("is-invalid")
+    }
+
+    // set released_at field to datetime without zone
+    const fullUtcDate = new Date(`${this.dateTarget.value} ${this.timeTarget.value} UTC`)
+    if (isNaN(fullUtcDate)) {
```

Comment:
oh i see, we just clear the field here if the date and time are "invalid"?

and then in the else case, we give an ISO but clear out the timezone "Z".

Reply:
Yeah, I think that was me getting "unexpected" results if you submit a weird looking string to Rails.  So I opted to just clear it out on the frontend.